source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx

```diff
@@ -58,6 +60,36 @@ export const DocumentDetailForm: React.FunctionComponent<
     [props.documentTypes],
   );
 
+  const [documentTypePurpose, setDocumentTypePurpose] = useState(null);
+
+  const matchDocumentType = useCallback(
+    (documentTypeId: number) => {
+      const purpose = props.documentTypes.find(
+        x => x.id === Number(documentTypeId),
+      )?.documentTypePurpose;
+
+      setDocumentTypePurpose(purpose);
```

Comment:
this will result in string | undefined | null I believe, do you want string | null here instead?